### PR TITLE
UseCase refactor

### PR DESCRIPTION
**Pull request by kin0992**
The goal of this PR is to simplify the use of the UseCases, trying to create a single UseCase which persists the record.

### CODE REVIEW

File: src/domain/__tests__/ConsumeEventStreamRecord.tests.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { makeConsumeEventStreamRecord } from '../ConsumeEventStreamRecord';
```
Comment (kin0992) [inline src/domain/__tests__/ConsumeEventStreamRecord.tests.ts:1]: If we don't want to repeat every time the invocation of the test, we can evaluate to use `jest-each` and write tests like this:
```
describe('makeConsumeEventStreamRecord', () => {
  each([
    {
      title: 'should return an empty array',
      input: data.consumeEventStreamRecord.input,
      records: [],
      expected: {
        statusCode: 200,
        returned: [],
      },
    },
    {
      title: 'should return an event',
      input: data.consumeEventStreamRecord.input,
      records: [data.newNotificationRecord],
      expected: {
        statusCode: 200,
        returned: data.consumeEventStreamResponse.returned,
      },
    },
...
  ]).it('$title', ({ input, records, expected }) => {
    const actual = makeConsumeEventStreamRecord(data.makeTestSystemEnv())(input)(records);
    expect(actual.output.statusCode).toStrictEqual(expected.statusCode);
    expect(actual.output.returned).toStrictEqual(expected.returned);
  });
});
```

Comment (datalek) [reply src/domain/__tests__/ConsumeEventStreamRecord.tests.ts:1]: Sincerely I'm not a big fun of `each`, the types are erased.

Comment (kin0992) [reply src/domain/__tests__/ConsumeEventStreamRecord.tests.ts:1]: That's true! I see an advantage to have only one place where add the new tests (in the each part), but the drawback is what you said.

**Review by datalek — APPROVED** [review]
🥇

**Pull request merged**